src/Hydration.js

```diff
@@ -1,25 +1,58 @@
 class Hydration {
-  constructor(hydrationData) {
-    this.hydrationData = hydrationData;
+  constructor(hydrationData, user) {
+    this.userId = user.id;
+    this.hydrationData = this.getHydrationData(hydrationData);
   }
-
-  calculateAverageOunces(id) {
-    const perDayUserHydration = this.hydrationData.filter((data) => id === data.userID);
-    return perDayUserHydration.reduce((sumSoFar, data) => sumSoFar += data.numOunces, 0) / perDayUserHydration.length;
+  
+  getHydrationData(hydrationData) {
+    let filteredData = hydrationData.filter(data => {
+      return this.userId === data.userId
+    })
+      return filteredData.map( data => {
+       return {date: data.date, numOunces: data.numOunces}
+    })
   }
-
-  calculateDailyOunces(id, date) {
-    const findOuncesByDate = this.hydrationData.find((data) => id === data.userID && date === data.date);
-    return findOuncesByDate.numOunces;
+  
+  calculateAverageOunces() {
+    let totalOunces = this.hydrationData.reduce((total, data) => {
+      total += data.numOunces
+      return total 
+    }, 0)
+    return Math.round(totalOunces / this.hydrationData.length)
   }
 
-  calculateFirstWeekOunces(userRepo, id) {
-    return userRepo.getFirstWeek(id, this.hydrationData).map((data) => `${data.date}: ${data.numOunces}`);
-  }
-
-  calculateRandomWeekOunces(date, id, userRepo) {
-    return userRepo.getWeekFromDate(date, id, this.hydrationData).map((data) => `${data.date}: ${data.numOunces}`);
-  }
+  calculateDailyOunces(date) {
+     let daily = this.hydrationData.find(data => {
+     return date === data.date
+     })
+     if(daily === undefined) 
+       return "Sorry there is no hydration data for that date"
+     return daily.numOunces
+   }
+  
+  calculateFirstWeekOunces(date) {
+    let weekData = this.findGivenWeek(date);
+      let weekHydration = weekData.reduce((wholeWeek, day) => {
+        wholeWeek.push(day)
+        return wholeWeek
+      }, [])
+      return weekHydration
+    }
+  
+    findGivenWeek(date) {
+      let sorted = this.hydrationData.sort((a, b) => {
+        return Date.parse(a.date) - Date.parse(b.date)
+      })
+        let endDate = sorted.findIndex(data =>{
```

Comment:
Looks good, let's make sure we are following our style guide indenting protocols (Line 11, calculateDailyOunces() on line 24, and line 35 and 39 inside calculateFirstWeekOunces())